python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -12,13 +12,23 @@
 
 @experimental
 class EcsRunLauncher(RunLauncher, ConfigurableClass):
-    def __init__(self, inst_data=None, task_definition=None, container_name="run"):
+    def __init__(
+        self,
+        inst_data=None,
+        task_definition=None,
+        container_name="run",
+        secrets=None,
+        secrets_tag="dagster",
+    ):
         self._inst_data = inst_data
         self.ecs = boto3.client("ecs")
         self.ec2 = boto3.resource("ec2")
+        self.secrets_manager = boto3.client("secretsmanager")
```

Comment:
is this something that users will always have available? if not, we could also load it only if secrets and secrets_tag are non-empty

Reply:
You can create the client with boto3 even without creds - nothing will fail until you try to use the client. So I don't think we need to defer initialization here.